source/glbinding/include/glbinding/Logging.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <thread>
+#include <fstream>
+
+#include <glbinding/AbstractValue.h> 
+#include <glbinding/callbacks.h>
+
+
+namespace glbinding 
+{
+
+template <typename T>
+class RingBuffer;
+
+
+class GLBINDING_API Logging
+{
+
+public:
+    static void start();
+    static void start(const std::string & filepath);
+    static void stop();
+    static void pause();
+    static void resume();
+
+    using BufferType = FunctionCall*;
+    static void log(BufferType  call);
+
+
+    using TailIdentifier = unsigned int;
+    static TailIdentifier addTail();
+    static void removeTail(TailIdentifier);
+    static const std::vector<BufferType>::const_iterator cbegin(TailIdentifier key);
+    static bool valid(TailIdentifier key, const std::vector<BufferType>::const_iterator & it);
+    static const std::vector<BufferType>::const_iterator next(TailIdentifier key, const std::vector<BufferType>::const_iterator & it);
+    static void release(TailIdentifier key, const std::vector<BufferType>::const_iterator & it);
+    static unsigned int size(TailIdentifier key);
+
+private:
+    Logging() = delete;
+    ~Logging() = delete;
+
+private:
+    static bool s_stop;
+    static bool s_persisted;
+    static std::mutex s_lockfinish;
+    static std::condition_variable s_finishcheck;
+
+
+    using FunctionCallBuffer = glbinding::RingBuffer<BufferType>;
+    static FunctionCallBuffer s_buffer;
```

Comment:
It seems like this code isn't prepared for multi-context and multi-threaded usage of glbinding.
It should be documented and maybe there should be an issue stating what is needed to extend your solution towards this.